File: Python/DayOfTheProgrammer.py
import datetime

def dayOfProgrammer(year):
    if year == 1918:
        # Special case for 1918
        return "26.09.1918"
    elif year < 1918:
        # Julian calendar before 1918
        return "12.09.%d" % year if year % 4 == 0 else "13.09.%d" % year
    else:
        # Gregorian calendar from 1918 onwards
        return "12.09.%d" % year if year % 400 == 0 or (year % 4 == 0 and year % 100 != 0) else "13.09.%d" % year

# Sample test case
year = 2021
expected_output = "13.09.2021"
actual_output = dayOfProgrammer(year)
print("Test passed" if actual_output == expected_output else "Test failed")

File: Java/DayOfTheProgrammer.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class DayOfTheProgrammer {
    public static String dayOfProgrammer(int year) {
        LocalDate date;
        if (year == 1918) {
            // Special case for 1918
            date = LocalDate.of(year, 9, 26);
        } else if (year < 1918) {
            // Julian calendar before 1918
            date = (year % 4 == 0) ? LocalDate.of(year, 9, 12) : LocalDate.of(year, 9, 13);
        } else {
            // Gregorian calendar from 1918 onwards
            date = ((year % 400 == 0) || ((year % 4 == 0) && (year % 100 != 0))) ? LocalDate.of(year, 9, 12) : LocalDate.of(year, 9, 13);
        }
        return date.format(DateTimeFormatter.ofPattern("dd.MM.yyyy"));
    }

    public static void main(String[] args) {
        // Test case
        int year = 2021;
        String expectedOutput = "13.09.2021";
        String actualOutput = dayOfProgrammer(year);
        System.out.println(actualOutput.equals(expectedOutput) ? "Test passed" : "Test failed");
    }
}